### ISSUE TITLE
run example failed

### ISSUE THREAD

**Issue by better629** [bytedance/byteps#41]
**Describe the bug**
installed byteps with gcc4.9 and tensorflow=1.11.0

when i run `python3 keras_mnist.py` 
it occurs that 
byteps-0.1.0-py3.6-linux-x86_64.egg/byteps/tensorflow/c_lib.cpython-36m-x86_64-linux-gnu.so: undefined symbol: _ZNK10tensorflow8OpKernel4nameB5cxx11Ev

**Environment (please complete the following information):**
 - OS: ubuntu
 - GCC version: gcc49
 - CUDA and NCCL version: cuda 10, nccl 2.4
 - Framework (TF, PyTorch, MXNet): tf = 1.11.0  keras = 2.2.4

**Additional context**

**Comment by changlan** [comment]
Looks like it is an ABI issue. Would you mind sharing the following information:
1. Did you use our prebuilt Docker image, or did you build BytePS yourself?
2. Did you install tensorflow=1.11.0 from PIP?

**Comment by better629** [comment]
> Looks like it is an ABI issue. Would you mind sharing the following information:
> 
> 1. Did you use our prebuilt Docker image, or did you build BytePS yourself?
> 2. Did you install tensorflow=1.11.0 from PIP?

1. build from source
2. pip3 install tensorflow==1.11.0

**Comment by bobzhuyb** [comment]
TF 1.11 and 1.14 should not have problems. This is a linking problem. It seems that when you run byteps, it is searching symbols in a different tf version than when you build it. 

Do you have multiple TF installed on your machine?

You can refer to similar issues for Horovod. https://github.com/horovod/horovod/issues/431#issuecomment-468191367

**Comment by better629** [comment]
Yes, i know it's a link problem.
The machine installed conda, and has multi env, so it does have multi tf. I will try the issue your offered.

**Comment by better629** [comment]
Ok, I describe the pipeline detailly.
The System is ubuntu 18.04, original gcc version is 7.9
The tensorflow is installed by pip in anaconda, and maybe compile by gcc 7.9.

The byteps needs gcc 4.9, ,so I change it to 4.9. And There are two symbol in libtensorflow_framework.so in tf.
`_ZNK10tensorflow8OpKernel4nameB5cxx11Ev`
`ZNK10tensorflow8OpKernel4nameEv`.

The `ZNK10tensorflow8OpKernel4nameEv` is from libtensorflow_framework.so, and 
```
nm /data/anaconda3/lib/python3.6/site-packages/tensorflow/libtensorflow_framework.so | grep ZNK10tensorflow8OpKernel4nameEv
0000000000459f90 T _ZNK10tensorflow8OpKernel4nameEv
```

So, if i compile byteps using gcc 4.9 with `/data/anaconda3/bin/python3 setup.py install`. The output library `c_lib.cpython-36m-x86_64-linux-gnu.so` contains `_ZNK10tensorflow8OpKernel4nameB5cxx11Ev`. It's a link problem. So I `export LD_LIBRARY_PATH=/data/anaconda3/lib/python3.6/site-packages/tensorflow/:$LD_LIBRARY_PATH` to indicate the tf .so path.  Then, the symbol becomes 
```
nm build/lib.linux-x86_64-3.6/byteps/tensorflow/c_lib.cpython-36m-x86_64-linux-gnu.so | grep ZNK10tensorflow8OpKernel4nameB5cxx11Ev
                 U _ZNK10tensorflow8OpKernel4nameB5cxx11Ev
```
**and `/data/anaconda3/bin/python3 -c "import byteps.keras as bps"` is fine. But when i run `/data/anaconda3/bin/python3 keras_mnist.py`, it occurs a `core dump` which i don't know where is the core file.**

Question:
should I compile the tensorflow with gcc4.9 at the begining???

**Comment by bobzhuyb** [comment]
@better629 The general suggestion is to use the same gcc version for BytePS and TF. If you are sure that your TF was built with gcc 7.9, I believe you can build BytePS using 7.9, too.

In other issues, we recommend users to use 4.9 because TF in python pip was usually built with gcc 4.

**Comment by better629** [comment]
> @better629 The general suggestion is to use the same gcc version for BytePS and TF. If you are sure that your TF was built with gcc 7.9, I believe you can build BytePS using 7.9, too.
> 
> In other issues, we recommend users to use 4.9 because TF in python pip was usually built with gcc 4 (though I am not sure about anaconda).

Yes, The pip tf built with gcc 7.9 was my guess. So I did compile the byteps with gcc 7.9. But also occurs the symbol problem.
The above means that I compiled the byteps, but `core dump` when it runs.
